rlog.go

```diff
@@ -80,30 +81,45 @@ type filter struct {
 	Level   int
 }
 
-// rlogConfig captures the entire configuration of rlog, as supplied by a user,
-// for example via environment variables.
+// rlogConfig captures the entire configuration of rlog, as supplied by a user
+// via environment variables and/or config files. This still requires checking
+// and translation into more easily used config items. All values therefore are
+// stored as simple strings here.
 type rlogConfig struct {
-	logLevel       string // What log level. String, since filters are allowed
-	traceLevel     string // What trace level. String, since filters are allowed
-	logTimeFormat  string // The time format spec for date/time stamps in output
-	logFile        string // Name of logfile
-	logStream      string // Name of logstream: stdout, stderr or NONE
-	logTimeDate    bool   // Flag to determine if date/time is logged at all
-	showCallerInfo bool   // Flag to determine if caller info is logged
+	logLevel        string // What log level. String, since filters are allowed
+	traceLevel      string // What trace level. String, since filters are allowed
+	logTimeFormat   string // The time format spec for date/time stamps in output
+	logFile         string // Name of logfile
+	confFile        string // Name of config file
+	logStream       string // Name of logstream: stdout, stderr or NONE
+	logNoTime       string // Flag to determine if date/time is logged at all
+	showCallerInfo  string // Flag to determine if caller info is logged
+	confCheckInterv string // Interval in seconds for checking config file
 }
 
-// The configuration items in rlogConfig are what is supplied by the user (for
-// example in environment variables). We interpret this and produce
-// pre-processed configuration values, which are stored in those variables
-// below.
-var settingShowCallerInfo bool   // whether we log caller info
-var settingDateTimeFormat string // flags for date/time output
-var settingLogFile string        // logfile name
-
-var logWriterStream *log.Logger // the first writer to which output is sent
-var logWriterFile *log.Logger   // the second writer to which output is sent
-var logFilterSpec *filterSpec   // filters for log messages
-var traceFilterSpec *filterSpec // filters for trace messages
+// We keep a copy of what was supplied via environment variables, since we will
+// consult this every time we read from a config file. This allows us to
+// determine which values take precedence.
+var configFromEnvVars rlogConfig
```

Comment:
Most of this could be a single `var (` block instead of putting `var` on each item.